modules/gdnavigation/navigation_mesh_generator.cpp

```diff
@@ -53,15 +60,15 @@
 #include "modules/gridmap/grid_map.h"
 #endif
 
-EditorNavigationMeshGenerator *EditorNavigationMeshGenerator::singleton = NULL;
+NavigationMeshGenerator *NavigationMeshGenerator::singleton = NULL;
 
-void EditorNavigationMeshGenerator::_add_vertex(const Vector3 &p_vec3, Vector<float> &p_verticies) {
+void NavigationMeshGenerator::_add_vertex(const Vector3 &p_vec3, Vector<float> &p_verticies) {
 	p_verticies.push_back(p_vec3.x);
 	p_verticies.push_back(p_vec3.y);
 	p_verticies.push_back(p_vec3.z);
```

Comment:
You may want to resize + 3 first and set the 3 last values instead, `Vector` calls `realloc` on every `push_back`

Reply:
This is a problem of the Vector that doesn't have a proper a reserve mechanism. I prefer to not take into account locally, so when it will be improved I can benefit from it automatically.